core/domain/activity_validators.py

```diff
@@ -0,0 +1,86 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validators for activity models."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from constants import constants
+from core.domain import activity_domain
+from core.domain import base_model_validators
+from core.platform import models
+import feconf
+
+(collection_models, exp_models,) = models.Registry.import_models([
+    models.NAMES.collection, models.NAMES.exploration])
+
+
+class ActivityReferencesModelValidator(
+        base_model_validators.BaseModelValidator):
+    """Class for validating ActivityReferencesModels."""
+
+    @classmethod
+    def _get_model_id_regex(cls, unused_item):
+        # Valid id: featured.
+        regex_string = '^%s$' % '|'.join(
```

Comment:
if we write: ^some|thing$ 
So will it math `somthing` (not `e`)?

Reply:
No either some will match or thing. See:

<img width="1414" alt="Screenshot 2020-08-24 at 6 13 57 PM" src="https://user-images.githubusercontent.com/15226041/91046303-ce7a5400-e635-11ea-933f-24ea5c60b601.png">

Reply:
Discussed offline - brackets are required. Updated.